woocommerce/woocommerce
# Allow extensions in Add Products task

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





As part of the project to allow canonical WooCommerce extensions to surface themselves as part of the WooCommerce onboarding tasklist. This PR adds a filter to allow for such extensions to add themselves to the task-list.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install this WooCommerce Gift-Cards pre-release version
2. On the Product task list (`/wp-admin/admin.php?page=wc-admin&task=products`) you should see "Gift Card" as an option at the end of the list (hidden under the "view more" link)
3. Clicking it should create a new gift card product (This step isn't directly being tested as the primary work involved is in the gift cards plugin)



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Add support for extending product types on onboarding task list

#### Comment 

</details>
